drivers/i2c/CMakeLists.txt

```diff
@@ -33,6 +32,19 @@ zephyr_library_sources_ifdef(CONFIG_I2C_STM32_V2
 	i2c_ll_stm32.c
 	)
 
+if(CONFIG_I2C_DW)
+  zephyr_library_sources(i2c_dw.c)
+  foreach(NUM RANGE 0 7)
+    if(CONFIG_I2C_${NUM})
```

Comment:
This is apparently making the Kconfig checks unhappy, @ulfalizer any ideas?

Reply:
I'll put in a PR to the ci-tools to clear up the warning.

Reply:
Yeah, might have to whitelist that one for now.

If more cases like this show up, then maybe the token pasting false positive fix that @Sizurka added could be wrangled into detecting this case too.

Reply:
Maybe something like `r"\bCONFIG_[A-Z0-9_]+\b(?!$|\s*##)"` would work (see https://github.com/zephyrproject-rtos/ci-tools/commit/05b727a477612f27d8ba45edde3e76f80ae4da3d), adding `$` to negative lookahead.

Reply:
Going through the regex right now and whitelisting some other stuff as well (`CONFIG_$VAR$` and `CONFIG_FOO_*`, last one for comments).

If `ci-tools` was in the same repo, I could tweak things in the same PR, but no such luck...

Reply:
PR: https://github.com/zephyrproject-rtos/ci-tools/pull/65

Reply:
Thanks @ulfalizer .. I really didn't want to mess with that regex.